packages/design-picker/src/components/mshots-image/index.tsx

```diff
@@ -35,57 +45,76 @@ export function mshotsUrl( url: string, options: MShotsOptions, count = 0 ): str
+	const [ loadedSrc, setLoadedSrc ] = useState( '' );
+	const [ count, setCount ] = useState( 0 );
+
+	useEffect( () => {
+		const img = new Image();
+		const srcUrl = mshotsUrl( src, options, count );
+		img.src = srcUrl;
+		img.onload = () => {
+			// Detect default image (Don't request a 400x300 image).
+			if ( img.naturalWidth !== 400 || img.naturalHeight !== 300 ) {
+				setLoadedSrc( srcUrl );
+			} else {
+				// Only refresh 10 times
+				count < MAXTRIES &&
+					// Triggers a target.src change with increasing timeouts
+					setTimeout( () => setCount( count + 1 ), count * 500 );
+			}
+		};
+	}, [ src, count ] );
+
+	return loadedSrc;
+};
+
+// For hover-scroll, we use a div with a background image (rather than an img element)
+// in order to use transitions between `top` and `bottom` on the
+// `background-position` property.
+// The "normal" top & bottom properties are problematic individually because we
+// don't know how big the images will be, and using both gets the
+// right positions but with no transition (as they're different properties).
 const MShotsImage = ( {
 	url,
 	'aria-labelledby': labelledby,
 	alt,
 	options,
+	scrollable = false,
 }: MShotsImageProps ): JSX.Element => {
-	const [ count, setCount ] = React.useState( 1 );
-	const [ visible, setVisible ] = useState( false );
+	const [ visible, setVisible ] = useState( true );
 	const [ opacity, setOpacity ] = useState( 0 );
 
-	const src = mshotsUrl( url, options, count );
+	const src = useMshotsUrl( url, options );
+	const backgroundImage = src && `url( ${ src } )`;
 
 	// Hide the images while they're loading if src changes (e.g. when locale is switched)
 	useLayoutEffect( () => {
 		// Opacity is used for fade in on load
 		// Visible is used to hide the image quickly when swapping languages
-		setVisible( false );
-		setOpacity( 0 );
-	}, [ src, setVisible, setOpacity ] );
-
-	return (
-		<div className="mshots-image__container">
-			{ ! visible && <div className="mshots-image__loader"></div> }
-			<img
-				className={ classnames(
-					'mshots-image',
-					visible ? 'mshots-image-visible' : 'mshots-image-hidden'
-				) }
-				style={ { opacity: opacity } }
-				alt={ alt }
-				aria-labelledby={ labelledby }
-				src={ src }
-				onLoad={ ( e ) => {
-					// Test mshots h value matches the desired image
-					// The default image (https://s0.wp.com/mshots/v1/default) is around 400x300 px h
-					// but sometimes slightly off (e.g. h: 299.99)
-					if ( e.currentTarget.naturalHeight !== options.h ) {
-						// Only refresh 10 times
-						if ( count < MAXTRIES ) {
-							// Triggers a target.src change and image refresh @ 500ms, 1000ms, 1500ms...
-							setTimeout( () => setCount( count + 1 ), count * 500 );
-						}
-					} else {
-						// Show the image / hide loader
-						setVisible( true );
-						// Fade in
-						setOpacity( 1 );
-					}
-				} }
-			/>
-		</div>
+		setVisible( !! src );
+		setOpacity( src ? 1 : 0 );
```

Comment:
Now that `visibility` and `opacity` are derived entirely from the value of `src`, I wonder why these have to be in their own state variables 🤔 
i.e. remove the `useState()` hooks that define these variables, remove the `useLayoutEffect()`, and just have
```
const visibility = !! src;
const opacity = src ? 1 : 0;
```

Haven't tested this though, maybe I'm missing something important.

Reply:
Yeah, this is the 'pretty tricky' I mentioned in the description and is a regression. The images should fade in once they've loaded and the `useLayoutEffect()` is supposed to set the opacity to zero when the src is changed before a redraw so that it can trigger the transition to 1 when the image is ready, but I haven't been able to get it to work properly yet.

I'm ready to punt this to a followup pr now (and/or ask for help: Do you see a good way to fade the image in?).

Reply:
Personally I think a follow up PR is fine, even if it means there's no fade in for a bit in production.

> Do you see a good way to fade the image in?

Would a CSS animation do the job? (the kind that uses `@keyframes`) They can be made to be none repeating.

Attach the fade in animation to a class, then add the class to the DOM element when `src` is truthy. When the image loads, `src` turns truthy, and React will add the CSS class, which fades in the image.
This might make the image fade in regardless of whether the image is already loaded or not though, so may need a way to only add the class if it's not the first render?